legal-api/src/legal_api/services/filings/validations/agm_extension.py

```diff
@@ -52,12 +51,12 @@ def validate(legal_entity: LegalEntity, filing: Dict) -> Optional[Error]:
     return None
 
 
-def first_agm_validation(legal_entity: LegalEntity, filing: Dict) -> list:
+def first_agm_validation(business: any, filing: Dict) -> list:
     """Validate filing for first AGM Extension."""
     msg = []
 
     has_ext_req_for_agm_year = get_bool(filing, f"{AGM_EXTENSION_PATH}/extReqForAgmYear")
-    founding_date = LegislationDatetime.as_legislation_timezone_from_date(legal_entity.founding_date).date()
+    founding_date = LegislationDatetime.as_legislation_timezone_from_date(business.founding_date).date()
```

Comment:
I don't think AlternateName has a `founding_date` attribute, and will likely throw an AttributeError if the business isn't a LegalEntity here. There's also a few other attributes in this PR that I am not sure if they are valid for AlternateName, so you may need to double check: `last_ar_date`, `last_agm_date`, `naics_code`, `dissolution_date`, `naics_description`, `restoration_expiry_date`.

Reply:
Think AlternateName doesn't have AGM Extension filing (and some other filings) and it will be blocked before going to validation. That's why I don't add checks for them.